systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -1069,6 +1075,333 @@ void testRestoreOffsetsInConsumerGroup(ExtensionContext extensionContext) {
+                .withKafkaUsername(userSource.getMetadata().getName())
+                .withMessageCount(messagesCount)
+                .withListenerName(kafkaTlsScramListener)
+                .build();
+
+        LOGGER.info("[DEFAULT] Sending messages to - topic {}, cluster {} and message count of {}",
+                topicSourceName, kafkaClusterSourceName, messagesCount);
+        int sent = internalKafkaClient.sendMessagesTls();
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+
+        LOGGER.info("[DEFAULT] Changing to target - topic {}, cluster {}, user {}", topicTargetName, kafkaClusterTargetName, userTarget.getMetadata().getName());
+        internalKafkaClient = internalKafkaClient.toBuilder()
+                .withTopicName(topicTargetName)
+                .withClusterName(kafkaClusterTargetName)
+                .withKafkaUsername(userTarget.getMetadata().getName())
+                .build();
+
+        LOGGER.info("[DEFAULT] Now messages should be mirrored to target topic and cluster");
+        internalKafkaClient.checkProducedAndConsumedMessages(sent, internalKafkaClient.receiveMessagesTls());
+        LOGGER.info("[DEFAULT] Messages successfully mirrored");
+
+        LOGGER.info("[SCRAM-SHA] Going to change secrets for KafkaUser source/target.");
+        Map<String, String> mmSnapshot = DeploymentUtils.depSnapshot(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName));
+        LOGGER.info("[SCRAM-SHA] Change sha-password KafkaUser on KMM2 Source and make sure it rolled");
+        Secret passwordSource = new SecretBuilder()
+                .withNewMetadata()
+                .withName(kafkaUserSource)
+                .endMetadata()
+                .addToData("password", "c291cmNlLXBhc3N3b3Jk")
+                .build();
+        kubeClient().patchSecret(namespaceName, kafkaUserSource, passwordSource);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+
+        LOGGER.info("[SCRAM-SHA] Change KafkaUser sha-password on KMM2 Target");
+        Secret passwordTarget = new SecretBuilder()
+                .withNewMetadata()
+                .withName(kafkaUserTarget)
+                .endMetadata()
+                .addToData("password", "dGFyZ2V0LXBhc3N3b3Jk")
+                .build();
+        kubeClient().patchSecret(namespaceName, kafkaUserTarget, passwordTarget);
+        mmSnapshot = DeploymentUtils.waitTillDepHasRolled(namespaceName, KafkaMirrorMaker2Resources.deploymentName(clusterName), 1, mmSnapshot);
+        Map<String, String> kafkaSourcePods = StatefulSetUtils.ssSnapshot(namespaceName, KafkaResources.kafkaStatefulSetName(kafkaClusterSourceName));
+        Map<String, String> kafkaTargetPods = StatefulSetUtils.ssSnapshot(namespaceName, KafkaResources.kafkaStatefulSetName(kafkaClusterSourceName));
```

Comment:
Why are you getting snapshots of `kafkaClusterSource` twice?

Reply:
Second should have been `kafkaClusterTargetName`. Nice eyes! ;-)